core/variant/variant.cpp

```diff
@@ -835,17 +835,6 @@ bool Variant::operator==(const Variant &p_variant) const {
 	return hash_compare(p_variant);
 }
 
-bool Variant::operator!=(const Variant &p_variant) const {
-	// Don't use `!hash_compare(p_variant)` given it makes use of OP_EQUAL
```

Comment:
This note matters, so I don't think this is a valid change

Reply:
It's not a requirement, no; if need be, this can be the exception to the implicit conversion. With that said, all the existing tests pass, and I don't *think* there's any Variant operation where `x != y` and `!(x == y)` produce different values. Still worth looking into for verification, but I'm skeptical of how much weight this note has.

Reply:
Ultimately decided to not revert, as there doesn't appear to be any scenario where a discrepancy between `OP_EQUAL` and `!OP_NOT_EQUAL` exists. If such a discrepancy **does** exist somewhere, the discrepancy *itself* should be what's changed.